### PR TITLE
Get to the point of basic syscalls working in the native ABI

### PR DESCRIPTION
**Pull request by bsdjhb**
This is able to run a purecap hello world using the native ABI.  sendsig and friends are still broken, and we will still need to audit / fix native system calls, but this lets us start doing more of that work in parallel.

### CODE REVIEW

File: sys/mips/mips/vm_machdep.c [hunk 1]
```diff
@@ -578,28 +583,49 @@ swi_vm(void *dummy)
...
 #endif
-#if defined (COMPAT_CHERIABI)
-	/*
-	 * XXX-AR: should cheriabi_set_user_tls just delegate to this
-	 * function?
-	 */
-	if (td->td_proc && SV_PROC_FLAG(td->td_proc, SV_CHERI))
-		panic("cpu_set_user_tls(%p) should not be called from CHERIABI\n", td);
+#ifdef COMPAT_FREEBSD64
+	if (!SV_PROC_FLAG(td->td_proc, SV_CHERI))
```
Comment (jrtc27) [inline sys/mips/mips/vm_machdep.c]: Perhaps better as `SV_PROC_FLAG(td->td_proc, SV_LP64 | SV_CHERI) == SV_LP64` like is done elsewhere in the kernel for freebsd64?

File: sys/mips/include/tls.h [hunk 2]
```diff
@@ -38,32 +38,34 @@
...
-#ifdef __CHERI_PURE_CAPABILITY__
+#if defined(__CHERI_PURE_CAPABILITY__) || \
+    (defined(_KERNEL) && __has_feature(capabilities))
 #define TLS_TP_OFFSET	0
 #define TLS_DTP_OFFSET	0
 #else
 #define TLS_TP_OFFSET	0x7000
 #define TLS_DTP_OFFSET	0x8000
+#endif
+#ifdef COMPAT_FREEBSD32
```
Comment (brooksdavis) [inline sys/mips/include/tls.h]: Does it actually make sense for these to be conditional on a kernel-only define when these aren't inside an `#ifdef _KERNEL`?

*he says as he spots the `COMPAT_CHERIABI` he probably wrote below...*

File: sys/mips/mips/pm_machdep.c [hunk 3]
```diff
@@ -617,32 +739,137 @@ exec_setregs(struct thread *td, struct image_params *imgp, uintcap_t stack)
...
+		 * want to use csig_csp at all?  Possibly csig_csp
+		 * should default to NULL...?
+		 */
+		csigp = &td->td_pcb->pcb_cherisignal;
+		csigp->csig_csp = td->td_frame->csp;
+		csigp->csig_default_stack = csigp->csig_csp;
+#ifdef CHERIABI_LEGACY_SUPPORT
+		csigp->csig_ddc = frame->ddc;
+#endif
+	} else
```
Comment (brooksdavis) [inline sys/mips/mips/pm_machdep.c]: Where does this block get close in the `CHERIABI_LEGACY_SUPPORT` case?

File: sys/mips/mips/vm_machdep.c [hunk 4]
```diff
@@ -72,6 +72,7 @@ __FBSDID("$FreeBSD$");
 
 #ifdef CPU_CHERI
```
Comment (brooksdavis) [inline sys/mips/mips/vm_machdep.c]: `__has_feature(capabilities)`?

File: sys/mips/include/tls.h (hunk 2 above)
Comment (bsdjhb) [reply sys/mips/include/tls.h]: I think it doesn't hurt.

File: sys/mips/mips/vm_machdep.c (hunk 4 above)
Comment (bsdjhb) [reply sys/mips/mips/vm_machdep.c]: Hmm, I had probably just not changed that out of trying to minimize the diff, but I can change this one.

File: sys/mips/mips/vm_machdep.c (hunk 1 above)
Comment (bsdjhb) [reply sys/mips/mips/vm_machdep.c]: Done

**Pull request merged**